src/components/Cluster/ClusterDetail/AppDetailsModal.js

```diff
@@ -42,41 +42,41 @@ const AppDetailsModal = props => {
     props.onClose();
   }
 
-  function dispatchDeleteAppConfig(app, clusterId, dispatch) {
-    dispatch(deleteAppConfig(app.metadata.name, clusterId))
+  function dispatchDeleteAppConfig(app, clusterId, dispatch, closeModal) {
+    return dispatch(deleteAppConfig(app.metadata.name, clusterId))
       .then(() => {
         return dispatch(loadApps(clusterId));
       })
       .then(() => {
-        onClose();
+        closeModal();
```

Comment:
I'm not sure about this change here, how did this ever work before :P I know I wrote it... but I see there is no onClose() being passed in to this function... weird.

Reply:
It works before `onClose()` is declared in this file, so it's in scope.
I just wanted it to be the same as the other `dispatch*` methods in the file, because I had some issues and I thought it would solve them, which they didn't. Left them like that for consistency, though

Reply:
Late to the party, sorry.

It worked because it is called on confirmation in `batchedClusterDeleteConfirmed()`. 

It would be nice if we clean these `then().catch()` blocks in components to avoid memory leaks, race conditions, etc. Generally we want them in actions

Reply:
A refactor is on the way, but that will come in a different PR 👍

Reply:
🎉